tests/w2w_test.py

```diff
@@ -190,9 +195,177 @@ def test_get_SW_BW():
+    src_data = rxr.open_rasterio(info['src_file'])[0, :, :]
+
+    lcz_arr = _get_lcz_arr(
+        src_data=src_data,
+        info=info
+    )
+
+    # Make sure shape is fine
+    assert lcz_arr.shape == (765, 1210)
+    # Only built LCZs labels should be in array.
+    assert list(np.unique(lcz_arr)) == [0, 2, 3, 5, 6, 8, 9]
+    # All other LCZs should be masked = 0.
+    assert np.sum(lcz_arr == 0) == 905938
+    # Type should be integere, to make sure lookup works
+    assert lcz_arr.dtype == np.int32
+
+
+def test_ucp_resampler_output_values_per_paramater():
+
+    info = {
+        'src_file': 'sample_data/lcz_zaragoza.tif',
+        'src_file_clean': 'testing/lcz_zaragoza_clean.tif',
+        'dst_gridinfo': 'testing/geo_em.d04_gridinfo.tif',
+        'BUILT_LCZ': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+    }
+
+    ucp_table = pd.read_csv(
+        'w2w/resources/LCZ_UCP_lookup.csv',
+         index_col=0
+    ).iloc[:17, :]
+
+    ucp_keys = [
+        'FRC_URB2D',
+        'MH_URB2D',
+        'STDH_URB2D',
+        'LB_URB2D',
+        'LF_URB2D',
+        'LP_URB2D'
+    ]
+
+    for ucp_key in ucp_keys:
+
+        if ucp_key == 'FRC_URB2D':
+            ucp_res = _ucp_resampler(
+                info=info,
+                ucp_key=ucp_key,
+                RESAMPLE_TYPE='average',
+                ucp_table=ucp_table,
+                FRC_THRESHOLD=0.2,
+            )
+        else:
+            ucp_res = _ucp_resampler(
+                info=info,
+                ucp_key=ucp_key,
+                RESAMPLE_TYPE='average',
+                ucp_table=ucp_table,
+            )
+        # Parameters should be on WRF grid size
+        assert ucp_res.shape == (1, 102, 162)
+
+        # Per parameter, check # max values and non-0 domain average
+        if ucp_key == 'FRC_URB2D':
+            assert np.sum(ucp_res.data == ucp_res.data.max()) == 1
+            assert approx(np.mean(ucp_res.data[ucp_res.data > 0])) == 0.428921
+        elif ucp_key == 'MH_URB2D':
+            assert np.sum(ucp_res.data == ucp_res.data.max()) == 2
+            assert approx(np.mean(ucp_res.data[ucp_res.data > 0])) == 1.1429937
+        elif ucp_key == 'STDH_URB2D':
+            assert np.sum(ucp_res.data == ucp_res.data.max()) == 2
+            assert approx(np.mean(ucp_res.data[ucp_res.data > 0])) == 0.29812142
+        elif ucp_key == 'LB_URB2D':
+            assert np.sum(ucp_res.data == ucp_res.data.max()) == 1
+            assert approx(np.mean(ucp_res.data[ucp_res.data > 0])) == 0.15104416
+        elif ucp_key == 'LF_URB2D':
+            assert np.sum(ucp_res.data == ucp_res.data.max()) == 1
+            assert approx(np.mean(ucp_res.data[ucp_res.data > 0])) == 0.07032267
+        elif ucp_key == 'LP_URB2D':
+            assert np.sum(ucp_res.data == ucp_res.data.max()) == 1
+            assert approx(np.mean(ucp_res.data[ucp_res.data > 0])) == 0.08072148
+
+        # Make sure no nans are present.
+        assert np.sum(np.isnan(ucp_res.data)) == 0
```

Comment:
```suggestion
@pytest.mark.parametrize(
    ('ucp_key', 'data_sum', 'data_mean'),
    (
        ('MH_URB2D', 2, 1.1429937),
        ('STDH_URB2D', 2, 0.29812142),
        ('LB_URB2D', 1, 0.15104416),
        ('LF_URB2D', 1, 0.07032267),
        ('LP_URB2D', 1, 0.08072148),
    ),
)
def test_ucp_resampler_output_values_per_paramater(
        ucp_key,
        data_sum,
        data_mean,
):
    info = {
        'src_file': 'sample_data/lcz_zaragoza.tif',
        'src_file_clean': 'testing/lcz_zaragoza_clean.tif',
        'dst_gridinfo': 'testing/geo_em.d04_gridinfo.tif',
        'BUILT_LCZ': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    }

    ucp_table = pd.read_csv('w2w/resources/LCZ_UCP_lookup.csv', index_col=0)

    ucp_res = _ucp_resampler(
        info=info,
        ucp_key=ucp_key,
        RESAMPLE_TYPE='average',
        ucp_table=ucp_table,
    )
    # Parameters should be on WRF grid size
    assert ucp_res.shape == (1, 102, 162)

    # Per parameter, check # max values and non-0 domain average
    assert np.sum(ucp_res.data == ucp_res.data.max()) == data_sum
    assert approx(np.mean(ucp_res.data[ucp_res.data > 0])) == data_mean
    # Make sure no nans are present.
    np.sum(np.isnan(ucp_res.data)) == 0


def test_ucp_resampler_output_values_per_paramater_frc_threshold():
    info = {
        'src_file': 'sample_data/lcz_zaragoza.tif',
        'src_file_clean': 'testing/lcz_zaragoza_clean.tif',
        'dst_gridinfo': 'testing/geo_em.d04_gridinfo.tif',
        'BUILT_LCZ': [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
    }

    ucp_table = pd.read_csv('w2w/resources/LCZ_UCP_lookup.csv', index_col=0)

    ucp_res = _ucp_resampler(
        info=info,
        ucp_key='FRC_URB2D',
        RESAMPLE_TYPE='average',
        ucp_table=ucp_table,
        FRC_THRESHOLD=0.2,
    )
    # Parameters should be on WRF grid size
    assert ucp_res.shape == (1, 102, 162)

    # Per parameter, check # max values and non-0 domain average
    assert np.sum(ucp_res.data == ucp_res.data.max()) == 1
    assert approx(np.mean(ucp_res.data[ucp_res.data > 0])) == 0.428921
    # Make sure no nans are present.
    np.sum(np.isnan(ucp_res.data)) == 0
```
I'd remove all the if/else logic, and then parametrize per key. The case with adding a `kwarg` I'd put in an extra test, since it is testing another freature